code/datums/quirks/negative_quirks/asthma.dm

```diff
@@ -0,0 +1,248 @@
+	RegisterSignal(quirk_holder, COMSIG_CARBON_POST_BREATHE, PROC_REF(holder_breathed))
+
+	time_next_attack_allowed = world.time + time_first_attack_can_happen
+
+/datum/quirk/item_quirk/asthma/remove()
+	. = ..()
+
+	current_attack?.cure()
+	UnregisterSignal(quirk_holder, COMSIG_CARBON_EXPOSED_TO_SMOKE, COMSIG_CARBON_LOSE_ORGAN, COMSIG_ATOM_REAGENTS_TRANSFERRED_TO, COMSIG_CARBON_POST_BREATHE)
+
+/datum/quirk/item_quirk/asthma/process(seconds_per_tick)
+	if(HAS_TRAIT(quirk_holder, TRAIT_STASIS))
+		return
+	if (quirk_holder.stat == DEAD)
+		return
+
+	var/mob/living/carbon/carbon_quirk_holder = quirk_holder
+	var/obj/item/organ/internal/lungs/holder_lungs = carbon_quirk_holder.get_organ_slot(ORGAN_SLOT_LUNGS)
+	if (!holder_lungs)
+		return
+
+	adjust_inflammation(-passive_inflammation_reduction * seconds_per_tick)
+
+	if (carbon_quirk_holder.has_reagent(/datum/reagent/toxin/histamine))
+		var/datum/reagent/toxin/histamine/holder_histimine = carbon_quirk_holder.reagents.get_reagent(/datum/reagent/toxin/histamine)
+		if (holder_histimine)
+			if (holder_histimine.overdosed) // uh oh!
+				if (SPT_PROB(15, seconds_per_tick))
+					to_chat(carbon_quirk_holder, span_boldwarning("You feel your neck swelling, squeezing on your windpipe more and more!"))
+				adjust_inflammation(histimine_OD_inflammation)
+			else
+				if (SPT_PROB(5, seconds_per_tick))
+					to_chat(carbon_quirk_holder, span_warning("You find yourself wheezing a little harder as your neck swells..."))
+				adjust_inflammation(histimine_inflammation)
+
+	if (carbon_quirk_holder.has_reagent(/datum/reagent/medicine/albuterol))
+		var/datum/reagent/medicine/albuterol/albuterol = carbon_quirk_holder.reagents.get_reagent(/datum/reagent/medicine/albuterol)
+		if (isnull(albuterol)) // sanity - couldve been purged
+			inhaled_albuterol = 0
+		else
+			inhaled_albuterol = min(albuterol.volume, inhaled_albuterol)
+
+		if (inhaled_albuterol > 0)
+			adjust_inflammation(-(albuterol_inflammtion_reduction * seconds_per_tick))
+
+	// asthma attacks dont happen if theres no client, because they can just kill you and some need immediate response
+	else if (carbon_quirk_holder.client && isnull(current_attack) && world.time > time_next_attack_allowed && SPT_PROB(chance_for_attack_to_happen_per_second, seconds_per_tick))
+		do_asthma_attack()
+
+/// Causes an asthma attack via infecting our owner with the attack disease. Notifies ghosts.
+/datum/quirk/item_quirk/asthma/proc/do_asthma_attack()
+	var/datum/disease/asthma_attack/typepath = pick_weight(GLOB.asthma_attack_rarities)
+
+	current_attack = new typepath
+	current_attack.infect(quirk_holder, make_copy = FALSE) // dont leave make_copy on FALSE. worst mistake ive ever made
+	RegisterSignal(current_attack, COMSIG_QDELETING, PROC_REF(attack_deleting))
+
+	notify_ghosts("[quirk_holder] is having an asthma attack: [current_attack.name]!", source = quirk_holder, action = NOTIFY_ORBIT, header = "Asthma attack!")
```

Comment:
To be brutally honest: perhaps the most pointless / nerdiest notify ghosts ever

Reply:
:(

Reply:
ok only severe ones will get a alert